### ISSUE TITLE
Wrong request for page creation

### ISSUE THREAD

**Issue by vitaliy-sk** [BoD/klibnotion#2]
Looks like a request for page creation by page ref is wrong.

Following code 

```java
RichTextList richTextList = new RichTextList();
        richTextList.text("Name", Annotations.DEFAULT);

        return pages.createPage(new PageReference(parentPage), richTextList, content)
```

produce request

```json
{
  "parent": {
    "type": "page_id",
    "page_id": "6bd77e46-914a-44dd-98a4-bd5f85d163a4"
  },
  "properties": {
    "title": {
      "rich_text": [
        {
          "text": {
            "content": "Name"
          }
        }
      ]
    }
  },
  "children": [
    {
      "object": "block",
      "type": "paragraph",
      "paragraph": {
        "text": [
          {
            "text": {
              "content": "Test paragrgaph"
            }
          }
        ]
      }
    }
  ]
}
```

it fails with errors

```json
{"object":"error","status":400,"code":"validation_error","message":"body failed validation. Fix one: body.parent.database_id should be defined, instead was `undefined`. body.properties.title should be an array or `undefined`, instead was `{\"rich_text\":[{\"text\":{\"content\":\"Name\"}}]}`."}
```

The correct request should look like this:

```json
{
    "parent": {
        "page_id": "6bd77e46-914a-44dd-98a4-bd5f85d163a4"
    },
    "properties": {
       "title": [
                {
                    "type": "text",
                    "text": {
                        "content": "Name"
                    }
                }
            ]
    },
    "children": [
        {
            "object": "block",
            "type": "paragraph",
            "paragraph": {
                "text": [
                    {
                        "type": "text",
                        "text": {
                            "content": "Test paragrgaph"
                        }
                    }
                ]
            }
        }
    ]
}
```

**Comment by BoD** [comment]
Thanks for reporting!

I've noticed this was wrong indeed, but this looks like a bug on the Notion API's side, because this makes the syntax different than from the "create page in database" case. Notion devs have been informed on the dedicated Slack, but so far this is not resolved on their side.  I'll leave this issue open for now.

**Comment by vitaliy-sk** [comment]
@BoD Looks like same syntax for DB is also working

```json
{
    "parent": {
        "database_id": "{{DATABASE_ID}}"
    },
    "properties": {
       "title": [
                {
                    "type": "text",
                    "text": {
                        "content": "Page title - Test 123"
                    }
                }
            ]
    },
    "children": [
        {
            "object": "block",
            "type": "paragraph",
            "paragraph": {
                "text": [
                    {
                        "type": "text",
                        "text": {
                            "content": "Lacinato kale is a variety of kale with a long tradition in Italian cuisine, especially that of Tuscany. It is also known as Tuscan kale, Italian kale, dinosaur kale, kale, flat back kale, palm tree kale, or black Tuscan palm."
                        }
                    }
                ]
            }
        }
    ]
}
```

**Comment by BoD** [comment]
Thanks for the insight! This is a bit odd - and doesn't match with the documentation.  I will wait for Notion's feedback on this, I still think it's a bug on their side.

**Comment by liquid-void** [comment]
Any update on this? 
I've got the same issue but when trying to create a very simple page in a database.
```
notionClient.pages.createPage(
        parentDatabase = DatabaseReference(DATABASE_ID),
        properties = PropertyValueList()
            .title("Name", "Gandalf")

    )
```
generates following body:
```
{
    "parent":  {
            "type":"database_id",
            "database_id":"bcxxxc8e-f2df-49ac-9d45-bexxcdxx6xc"
    },
    "properties":  {
            "Name":{
                    "title":  [ {
                                      "text":  {
                                            "content":"Gandalf"
                                       }  
                                 } ]
            }
    },
   "children":null
}
```
Gives following error:

> Caused by: io.ktor.client.features.ClientRequestException: Client request(https://api.notion.com/v1/pages) invalid: 400 . Text: "{"object":"error","status":400,"code":"validation_error","message":"body failed validation: body.parent.page_id should be defined, instead was `undefined`."}"

**Comment by BoD** [comment]
@liquid-void This is odd, for a database parent this should definitely work. It is known to fail for a page parent only.

Are you sure the DATABASE_ID you passed is correct?

Or maybe Notion changed something on their side that broke this?

I will try to have a look on my side with my workspace and tell you if I can reproduce the issue.

**Comment by liquid-void** [comment]
@BoD 
I just tested with Insomnia and curl.
And it makes no difference whether I provide this line
`            "type":"database_id",`
or not.

The issue is this line in the request body
`   "children":null`
I tried leaving it out and now the response is fine.

**Comment by BoD** [comment]
Thanks for the update, this is good information. Yes the `children: null` looks suspicious. I'll see what I can do.

**Comment by BoD** [comment]
@liquid-void  I've just published v1.5.1 which fixes this issue.

However the title is still ignored in the "parent is a page" case - but this on Notion API's side. 

Closing this issue for now as there is nothing I can do on the client side.
